### ISSUE TITLE
Cannot install firmware on MKS TFT28 V2

### ISSUE THREAD

**Issue by Buggy123** [bigtreetech/BIGTREETECH-TouchScreenFirmware#1827]
# Summary
I am trying to install this firmware on a MKS TFT28 V2 touchscreen. The TFT does not finish installing, it freezes at 100% updated for a indefinite duration (~15 minutes tested).

# Details
I followed the installation instructions carefully using the latest version of all files on the master branch. The SD card looks like this before installation:
![image](https://user-images.githubusercontent.com/16315400/114295476-792f6d80-9a73-11eb-8e4e-2c97c67550a1.png)

After inserting the SD card and powering on the touchscreen, it beeps once, and then begins updating. The update progress bar reaches 100% and then halts without any further progression:
![IMG_20210411_030313](https://user-images.githubusercontent.com/16315400/114295832-aed55600-9a75-11eb-8b4b-431462e33386.jpg)

If reset or power cycled, the touchscreen begins the update process again with the same results. If the sd card is removed, the touchscreen remains stuck on the boot screen instead.

# Additional Notes

The config.ini is slightly modified; I'm currently using the config_rrf.ini file (renamed), as it seemed to more closely match the settings necessary for my printer, although I don't know what a couple of the differing settings do. I also changed the number of hotends to 2, for the same reason. I have tested it with the normal config.ini file, and it gives the same results.

The white line seen in the above picture is a physical defect; it is always present and has nothing to do with the firmware.

I have let it sit for at least 15 minutes during testing to ensure it does not progress further after enough time.

For simplicity and troubleshooting, I have completely removed the touchscreen from the printer. It also has no peripherals of any sort attached; the only connection is a USB cable connected to my computer for power.

The touchscreen can successfully be updated with the native MKS firmware and functions normally while using it.

The SD card used is a 32gb SDXC microSD card in a sd-card adaptor, formatted fat32. While a maximum of 8gb is stated, I am assuming the card is not a issue; the stock firmware can be flashed successfully with the same card, and the touchscreen has otherwise shown no issue with reading it.

The printer is a BIBO 2, which uses a MKS Gen L V1.0 board and also comes with a MKS Wifi TFT V1.0 peripheral for the touchscreen.

**Comment by digant73** [comment]
MKS TFT28 V3.0 and v4.0 are supported. V2.0 is not supported

**Comment by oldman4U** [comment]
According to the config.ini, the following TFT are currently supported:
#--------------------------------------------------------------------
# Supported TFT Variants
#--------------------------------------------------------------------
#
# BIGTREE_TFT35_V1_0    / BIGTREE_TFT35_V1_1 / BIGTREE_TFT35_V1_2
# BIGTREE_TFT35_V2_0    / BIGTREE_TFT35_V3_0 / BIGTREE_TFT35_E3_V3_0
# BIGTREE_TFT35_B1_V3_0 / BIGTREE_TFT43_V3_0 / BIGTREE_TFT50_V3_0
# BIGTREE_TFT70_V3_0    / BIGTREE_TFT28_V1_0 / BIGTREE_TFT28_V3_0
# BIGTREE_TFT24_V1_1    / MKS_TFT32_V1_3     / MKS_TFT32_V1_4
# MKS_TFT28_V3_0        / MKS_TFT28_V4_0

Please help the community and close this ticket.

Thank you

**Comment by radek8** [comment]
Hello
if it is
MKS_TFT32_V1_3 compatible with MSK_TFT28_V3_0
and
MKS_TFT32_V1_4 compatible with MSK_TFT28_V4_0,
will not the MKS_TFT32_V1_2 display also be compatible with MKS_TFT28_V2_0?

Try using this file to update.
https://github.com/Uz45/BIGTREETECH-TouchScreenFirmware/blob/master/Copy%20to%20SD%20Card%20root%20directory%20to%20update/MKS_32_V1_2.26.x.bin

Also use icons from this store. They will correspond better with the bin file
https://github.com/Uz45/BIGTREETECH-TouchScreenFirmware/tree/master/Copy%20to%20SD%20Card%20root%20directory%20to%20update

**Comment by Buggy123** [comment]
That's better - it progresses to the custom update screen/progress bar, but it's not quite there yet. All of the text looks like this:

![IMG_20210411_204405 - Copy](https://user-images.githubusercontent.com/16315400/114327559-8c8c1880-9b07-11eb-83cb-889e1451bca0.jpg)

I'm guessing this is the "update failed" screen. I'm not sure what this issue is, exactly... something with the fonts, maybe? I don't suppose you recognize it?

**Comment by Buggy123** [comment]
Figured it out!

![IMG_20210411_223356](https://user-images.githubusercontent.com/16315400/114333517-7b96d380-9b16-11eb-94cc-62fba17820d6.jpg)

Turns out the installation for that older version is slightly different - you aren't supposed to rename the MKS folder. 
At first, the touch screen wasn't responsive - turns out I needed to trigger the calibration process by putting a blank "reset.txt" on the sd card. It seems fully functional now, though I haven't hooked it up to the printer yet.

Either way, thank you!
